Source/AliveLibAO/Abe.cpp

```diff
@@ -4455,9 +4456,150 @@ void Abe::State_16_HoistBegin_426E40()
     }
 }
 
+ALIVE_VAR(1, 0xA08628, unsigned __int8, byte_A08628, 0);
+
+void PSX_Prevent_Rendering_44FFB0()
+{
+    byte_A08628 = 1;
+}
+
 void Abe::State_17_HoistIdle_4269E0()
 {
-    NOT_IMPLEMENTED();
+    PathLine* pPathLine = nullptr;
+    FP hitX = {};
+    FP hitY = {};
+    const auto bCollision = InAirCollision_4019C0(&pPathLine, &hitX, &hitY, FP_FromDouble(1.8));
+    SetActiveCameraDelayedFromDir_401C90();
+    if (bCollision)
+    {
+        switch (pPathLine->field_8_type)
+        {
+            case eFloor_0:
+            case eBackGroundFloor_4:
+            case 32:
+            case 36:
+            {
+                field_A8_xpos = hitX;
+                field_AC_ypos = hitY;
+
+                MapFollowMe_401D30(1);
+
+                field_F4_pLine = pPathLine;
+
+                field_FC_current_motion = eAbeStates::State_18_HoistLand_426EB0;
+                field_E4_previous_motion = eAbeStates::State_17_HoistIdle_4269E0;
+
+                PSX_RECT rect = {};
+                VGetBoundingRect_418120(&rect, 1);
+                rect.y += 5;
+                rect.h += 5;
+
+                VOnCollisionWith_418080(
+                    { rect.x, rect.y },
+                    { rect.w, rect.h },
+                    ObjListPlatforms_50766C,
+                    1,
+                    (TCollisionCallBack) &BaseAliveGameObject::OnTrapDoorIntersection_401C10
+                );
+                break;
+            }
+        }
+        return;
+    }
+
+    PullRingRope* pPullRope = GetPullRope_422580();
+    field_160_pRope = pPullRope;
+    if (pPullRope)
+    {
+        if (pPullRope->Pull_454CB0(this))
```

Comment:
Isn't Pull virtual? 🤔

Reply:
It's not market as one, at least in AO, what do?

Reply:
its fine if it isn't virtual, must only be virtual in AE